src/Error.h

```diff
@@ -182,46 +182,49 @@ struct WarningReport final : ReportBase<WarningReport> {
 };
 
 /**
- * _halide_internal_diagnostic is used to implement our assertion macros
- * in such a way that the messages output for the assertion are only
- * evaluated if the assertion's value is false.
+ * The following three diagnostic macros are implemented such that the
+ * message is evaluated only if the assertion's value is false.
  *
  * This (regrettably) requires a macro to work, but has the highly desirable
  * effect that all assertion parameters are totally skipped (not ever evaluated)
  * when the assertion is true.
  *
- * The macro works by deferring the call to issue() until after the stream
+ * The macros work by deferring the call to issue() until after the stream
  * has been evaluated. This previously used a trick where ErrorReport would
  * throw in the destructor, but throwing in a destructor is UB in a lot of
  * scenarios, and it was easy to break things by mistake.
  */
-// clang-format off
-#define _halide_internal_diagnostic(condition, type, condition_string)  \
-    /* NOLINTNEXTLINE(bugprone-macro-parentheses) */ \
-    if (!(condition)) for (type _err; _err; _err.issue()) _err.init(__FILE__, __FUNCTION__, __LINE__, condition_string)
-// clang-format on
+/// @{
+#define _halide_error_impl(type)                                    \
+    for (Halide::Internal::ErrorReport<type> _err; 1; _err.issue()) \
+    /**/ _err.init(__FILE__, __FUNCTION__, __LINE__, nullptr)
 
-#define _halide_internal_error(type) \
-    _halide_internal_diagnostic(0, Halide::Internal::ErrorReport<type>, nullptr)
+#define _halide_assert_impl(condition, type) \
+    if (!(condition))                        \
+    /**/ _halide_error_impl(type)
 
-#define _halide_internal_assertion(condition, type) \
-    _halide_internal_diagnostic(condition, Halide::Internal::ErrorReport<type>, #condition)
```

Comment:
I see `#condition` is gone. Do we still get the condition printed in the new implementation?

Reply:
Oops